test/blackbox/PubSubWriterReader.hpp

```diff
@@ -355,9 +542,103 @@ class PubSubWriterReader
         return *this;
     }
 
+    size_t get_num_discovered_participants() const
+    {
+        return participant_listener_.get_num_discovered_participants();
+    }
+
+    size_t get_num_discovered_publishers() const
+    {
+        return participant_listener_.get_num_discovered_publishers();
+    }
+
+    size_t get_num_discovered_subscribers() const
+    {
+        return participant_listener_.get_num_discovered_subscribers();
+    }
+
+    void print_discovered_participants()
```

Comment:
These `print` functions are accessing data without protection. Protect them. Or remove them because are not using in any place.

Reply:
Also these functions are sorting a std::set.

Reply:
They were debugging methods and have been removed.